[Bug]: export_to_urdf drops the sign of negative-axis joints (*_ROT_NEG/*_PRISM_NEG exported with positive axis)

### Prerequisites

- [x] I have searched existing issues and discussions and could not find a duplicate.
- [x] I have read the relevant section of the documentation.
- [x] I can reproduce this on the latest `main` (or the most recent release).

### Bug summary

`KinematicsParams.export_to_urdf(...)` drops the **sign** of revolute/prismatic joint axes: any joint that cuRobo classifies as a `*_ROT_NEG` / `*_PRISM_NEG` type (i.e. a URDF joint authored with a negative axis such as `<axis xyz="-1 0 0"/>`) is re-emitted in the exported URDF with the **positive** axis (`1 0 0`), and the joint origin/limits are not compensated. The exported URDF is therefore **kinematically inconsistent** with the source: for the same joint value, FK on the exported URDF rotates the affected joints the **opposite** direction.

This surfaced via `curobo.viewer.ViserVisualizer` (which renders a robot that has `extra_links` by exporting cuRobo's internal model to a temp URDF): a dexterous hand with negative-axis finger joints rendered with the wrong articulation (fingers bending the wrong way), diverging ~27.5 mm at the fingertips from the source URDF, even though the source URDF and the original `dex_retargeting` URDF agree exactly (0.0 mm).

### Steps to reproduce

```python
# Minimal, self-contained: a 2-link robot with ONE revolute joint whose axis is NEGATIVE.
import tempfile, textwrap, xml.etree.ElementTree as ET
import torch
from curobo._src.robot.kinematics.kinematics import Kinematics, KinematicsCfg
from curobo._src.types.device_cfg import DeviceCfg

urdf = tempfile.mktemp(suffix=".urdf")
with open(urdf, "w") as f:
    f.write(textwrap.dedent('''\
        <?xml version="1.0"?>
        <robot name="neg_axis_demo">
          <link name="base_link"><inertial><mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/></inertial></link>
          <link name="link1"><inertial><mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/></inertial></link>
          <joint name="j1" type="revolute">
            <parent link="base_link"/><child link="link1"/>
            <origin xyz="0.1 0 0" rpy="0 0 0"/>
            <axis xyz="-1 0 0"/>
            <limit lower="-1.57" upper="1.57" effort="1" velocity="1"/>
          </joint>
        </robot>'''))

kin_dict = {
    "urdf_path": urdf, "base_link": "base_link", "tool_frames": ["link1"],
    "cspace": {"joint_names": ["j1"], "default_joint_position": [0.0],
               "null_space_weight": [1.0], "cspace_distance_weight": [1.0],
               "max_acceleration": 1.0, "max_jerk": 10.0},
}
model = KinematicsCfg.from_data_dict(
    kin_dict, device_cfg=DeviceCfg(device=torch.device("cuda"), dtype=torch.float32))
kin = Kinematics(model)

out = tempfile.mktemp(suffix=".urdf")
kin.config.kinematics_config.export_to_urdf(output_path=out, kinematics_parser=model.kinematics_parser)

src = ET.parse(urdf).getroot().find("joint").find("axis").get("xyz")
exp = ET.parse(out).getroot().find("joint").find("axis").get("xyz")
print("source   j1 axis:", src)   # -1 0 0
print("exported j1 axis:", exp)   # 1.0 0.0 0.0   <-- sign dropped
assert src.replace(" ", "") == exp.replace(" ", ""), \
    "export_to_urdf dropped the joint axis sign"
```

### Expected behavior

The exported URDF should be kinematically identical to the source. A joint authored with `<axis xyz="-1 0 0"/>` should export with `axis = -1 0 0` (or an equivalent representation that yields the same FK), so that applying the same joint values to the source and exported URDFs gives the same link poses.

### Actual behavior / error output

```text
source   j1 axis: -1 0 0
exported j1 axis: 1.0 0.0 0.0
AssertionError: export_to_urdf dropped the joint axis sign
```

The exported axis is `+1 0 0` instead of `-1 0 0`. Because the origin is unchanged, FK on the exported URDF rotates `j1` in the opposite direction for the same joint angle. (In our real case — a robotera_xhand1 hand — the `*_hand_thumb_rota_joint*` joints flipped `-1 0 0 → +1 0 0`, producing a 27.5 mm fingertip divergence; positive-axis joints exported unchanged and were correct.)

### cuRobo version + commit SHA

0.8.0.post1.dev31 @ 2790cc5

### Installation method

Source — CUDA 12 + PyTorch (`uv pip install .[cu12-torch]`)

### Kernel backend

cuda_core (default, runtime compilation)

### Python version

3.12.3

### PyTorch version (if installed)

2.11.0+cu128 / CUDA 12.8

### GPU / driver / CUDA toolkit

NVIDIA GeForce RTX 3090, driver 575.57.08, CUDA toolkit 12.8

### Operating system

Ubuntu 24.04.4 LTS, kernel 6.8.0-124-generic

### Isaac Sim version (if applicable)

N/A

### Additional context

> **Update (corrected root cause + fix posted as #677).** My original analysis below was wrong about *where* the sign is lost; the corrected explanation follows. The symptom, repro, and expected/actual above are unchanged.

**Root cause** — `UrdfRobotParser._add_link` (`curobo/_src/robot/parser/parser_urdf.py`, ~lines 289–300) does **not** assign the `*_ROT_NEG` / `*_PRISM_NEG` joint types for a negative URDF axis. Instead it classifies the joint as the **positive** type (`abs(joint_axis[i]) == 1` → `X_ROT` etc.) and stores the sign as a `-1` scale in `joint_offset_map`:

```python
elif joint_type == "revolute":
    if abs(joint_axis[0]) == 1:
        joint_type = JointType.X_ROT          # positive type, even for axis = -1 0 0
    ...
if joint_axis[0] == -1 or joint_axis[1] == -1 or joint_axis[2] == -1:
    joint_offset[0] = -1.0 * joint_offset[0]  # sign goes into joint_offset, not the type
    joint_axis = [abs(x) for x in joint_axis]
```

cuRobo's FK is therefore correct (it applies the `joint_offset` scale). But `KinematicsParams.export_to_urdf` (`kinematics_params.py`) emits the axis purely from the joint **type** (`_joint_type_to_urdf_type` → positive axis) and **ignores `joint_offset_map`**, so the exported axis loses the sign while the origin is unchanged → the FK divergence above. (The `joint_offset_map` docstring confirms it stores scalars "for mimic joints and negative axis joints.")

**Fix (posted as #677):** re-apply `sign(joint_offset_map[...])` to the exported axis at the joint-emission site in `export_to_urdf`, plus a parametrized round-trip regression test (±X/±Y/±Z: exported axis sign preserved AND FK matches the source URDF). DCO sign-off + SSH-signed.

--- a/curobo/_src/robot/types/kinematics_params.py
+++ b/curobo/_src/robot/types/kinematics_params.py
@@ -886,6 +886,16 @@ class KinematicsParams:
             # Convert to URDF joint type and axis
             urdf_joint_type, axis = _joint_type_to_urdf_type(joint_type_value)
 
+            # A joint authored with a negative URDF axis (e.g. <axis xyz="-1 0 0"/>) is
+            # stored by UrdfRobotParser as the POSITIVE joint type plus a -1 joint-offset
+            # scale (parser_urdf.py), so _joint_type_to_urdf_type returns the positive axis.
+            # Re-apply that stored sign here, otherwise the exported axis loses it and FK on
+            # the exported URDF rotates the opposite way for the same joint value.
+            if axis is not None and joint_idx >= 0:
+                offset_scale = float(self.joint_offset_map[2 * link_idx].cpu().numpy())
+                if offset_scale < 0:
+                    axis = -axis
+
             # Get fixed transform (origin)
             fixed_transform = self.fixed_transforms[link_idx].cpu().numpy()
             origin = np.eye(4)

--- a/curobo/tests/_src/robot/types/test_kinematics_params.py
+++ b/curobo/tests/_src/robot/types/test_kinematics_params.py
@@ -5,8 +5,10 @@
 """Integration tests for KinematicsParams using KinematicsLoader."""
 
 # Third Party
+import numpy as np
 import pytest
 import torch
+import yourdfpy
 
 from curobo._src.robot.kinematics.kinematics_cfg import KinematicsCfg
 
@@ -496,8 +498,6 @@ class TestExportToUrdfAxisSign:
 
     @pytest.mark.parametrize("axis", ["-1 0 0", "1 0 0", "0 -1 0", "0 0 -1"])
     def test_export_preserves_axis_sign_and_fk(self, axis, tmp_path, cuda_device_cfg):
-        import numpy as np
-        import yourdfpy
 
         src_path = tmp_path / "src.urdf"
         self._write_single_joint_urdf(src_path, axis)

--- a/curobo/tests/_src/robot/types/test_kinematics_params.py
+++ b/curobo/tests/_src/robot/types/test_kinematics_params.py
@@ -467,3 +467,67 @@ class TestKinematicsParamsEdgeCases:
             franka_kinematics_cfg.reference_link_spheres,
         )
 
+
+
+class TestExportToUrdfAxisSign:
+    """Regression for NVlabs/curobo#676: ``export_to_urdf`` must preserve the sign of a
+    joint axis. A joint authored with a negative URDF axis (e.g. ``<axis xyz="-1 0 0"/>``)
+    is stored by the parser as the positive joint type plus a ``-1`` joint-offset scale;
+    the export must re-apply that sign, else FK on the exported URDF rotates the wrong way.
+    """
+
+    @staticmethod
+    def _write_single_joint_urdf(path, axis: str) -> None:
+        path.write_text(
+            '<?xml version="1.0"?>\n'
+            '<robot name="axis_sign_demo">\n'
+            '  <link name="base_link"><inertial><mass value="1.0"/>'
+            '<inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/></inertial></link>\n'
+            '  <link name="link1"><inertial><mass value="1.0"/>'
+            '<inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/></inertial></link>\n'
+            '  <joint name="j1" type="revolute">\n'
+            '    <parent link="base_link"/><child link="link1"/>\n'
+            '    <origin xyz="0.1 0 0" rpy="0 0 0"/>\n'
+            f'    <axis xyz="{axis}"/>\n'
+            '    <limit lower="-1.57" upper="1.57" effort="1" velocity="1"/>\n'
+            '  </joint>\n'
+            '</robot>\n'
+        )
+
+    @pytest.mark.parametrize("axis", ["-1 0 0", "1 0 0", "0 -1 0", "0 0 -1"])
+    def test_export_preserves_axis_sign_and_fk(self, axis, tmp_path, cuda_device_cfg):
+        import numpy as np
+        import yourdfpy
+
+        src_path = tmp_path / "src.urdf"
+        self._write_single_joint_urdf(src_path, axis)
+
+        kin_dict = {
+            "urdf_path": str(src_path), "base_link": "base_link", "tool_frames": ["link1"],
+            "cspace": {
+                "joint_names": ["j1"], "default_joint_position": [0.0],
+                "null_space_weight": [1.0], "cspace_distance_weight": [1.0],
+                "max_acceleration": 1.0, "max_jerk": 10.0,
+            },
+        }
+        cfg = KinematicsCfg.from_data_dict(kin_dict, device_cfg=cuda_device_cfg)
+
+        out_path = tmp_path / "exported.urdf"
+        cfg.kinematics_config.export_to_urdf(
+            output_path=str(out_path), kinematics_parser=cfg.kinematics_parser
+        )
+
+        src_urdf = yourdfpy.URDF.load(str(src_path), load_meshes=False, build_scene_graph=True)
+        exp_urdf = yourdfpy.URDF.load(str(out_path), load_meshes=False, build_scene_graph=True)
+
+        # 1. The exported axis must match the source axis (up to sign-equal float parse).
+        want = np.array([float(x) for x in axis.split()])
+        got = np.array([float(x) for x in exp_urdf.joint_map["j1"].axis])
+        np.testing.assert_allclose(got, want, atol=1e-9)
+
+        # 2. FK round-trip: child link pose must match the source for a non-trivial angle.
+        q = 0.7
+        src_urdf.update_cfg(np.array([q])); exp_urdf.update_cfg(np.array([q]))
+        np.testing.assert_allclose(
+            exp_urdf.get_transform("link1"), src_urdf.get_transform("link1"), atol=1e-6
+        )

